### PR TITLE
Update to support flysystem 2@rc

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: src/Integration/Flysystem/S3/src/S3FilesystemV2.php [hunk 1]
```diff
@@ -91,31 +93,36 @@ class S3FilesystemV2 implements FilesystemAdapter
...
-            $this->client->getObject(
+            return $this->client->objectExists(
                 [
                     'Bucket' => $this->bucket,
                     'Key' => $this->prefixer->prefixPath($path),
                 ]
-            )->resolve();
-
-            return true;
+            )->isSuccess();
```
Comment (Nyholm) [inline src/Integration/Flysystem/S3/src/S3FilesystemV2.php:123]: This is non-blocking, right?

Comment (jderusse) [reply src/Integration/Flysystem/S3/src/S3FilesystemV2.php:126]: nop, this just returns if the object exist. 
To be blocking (waiting until a final state (either success or failure)) use `$waiter->wait() && $waiter->isSuccess()`

**Review by jderusse — APPROVED** [review]
LGTM

File: src/Integration/Flysystem/S3/src/S3FilesystemV2.php (hunk 1 above)
Comment (Nyholm) [reply src/Integration/Flysystem/S3/src/S3FilesystemV2.php:126]: Cool cool. 

Then it is all good. It should not be blocking.

**Comment by Nyholm** [comment]
Thank you

**Pull request merged**